### DISCUSSION TITLE
how to avoid camera notch on the whole game screen?

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#1966] [Q&A] [answered]
same as the title .. by default the whole screen appearing full including camera hole (notch) area.

**Comment by rh101** [comment]
You need to only draw within the safe area, which you can get via `Director::getSafeAreaRect()`.

**Comment by GamesStudioTwo** [comment]
yes use the vars.
    auto visibleSize = _director->getVisibleSize();
    auto origin = _director->getVisibleOrigin();
    auto safeArea = _director->getSafeAreaRect();
    auto safeOrigin = safeArea.origin;

They map out your visible screen size no matter how much overhang there is from your set resolution size.
so the safeArea witch is a nice edition AXMOL added is where you start and end on screen.

**Reply by meehjeevan** [reply]
No, I'm not saying to draw within safe area, I want the game screen to be under notch at top area kd the screen.
If I use safe area to draw myself, it will also avoid the rounded corner area of top and bottom too. I fixed this in my cocos2dx version, but I forgot where exactly I adjusted the things for it.

Plz check the pics.
It's looking like this now:
![received_1199623928144875.jpeg](https://github.com/axmolengine/axmol/assets/6400139/9ee1fd61-5ee3-4bee-89a0-2bd5f6c739f6)


But I want to make it look like this:
![received_968992341681886.jpeg](https://github.com/axmolengine/axmol/assets/6400139/b3cca77c-5f89-40b0-8deb-968828ccace3)

**Reply by rh101** [reply]
In that case, what you may be referring to is the Android [display cutouts](https://developer.android.com/develop/ui/views/layout/display-cutout).

> If I use safe area to draw myself, it will also avoid the rounded corner area of top and bottom too. I fixed this in my cocos2dx version, but I forgot where exactly I adjusted the things for it.

If you already had this working in a previous version of your app, then why don't you just compare the old project to the new? You should be able to spot all changes you have made.  For this, it's most likely a change in the `proj.android/app/AndroidManifest.xml` file.

**Reply by meehjeevan** [reply, marked as answer]
Ok I found it's being controlled from java codes and not from manifest.xml.

I changed WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES  to 
WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_NEVER;

And it worked.
